improver/cube_combiner.py

```diff
@@ -103,23 +103,55 @@ def _check_dimensions_match(cube_list):
                        "{} and {}".format(repr(cube_list[0]), repr(cube)))
                 raise ValueError(msg)
 
-    def process(self, cube_list, new_diagnostic_name, coords_to_expand=None):
+    @staticmethod
+    def _get_expanded_coord_names(cube_list):
```

Comment:
To fully test this functionality it would be good to have tests that show:
-  that if the scalar coordinates on the input cubes are identical no coordinates are returned
- that the correct coordinates are returned if the scalar coordinates have the same names but different values
 - that it works for more than 2 cubes

I think all of these except for the more than 2 cubes may be covered by the process unit tests. I would find it clearer if there were unit tests for this function separately, but if the process unit tests cover this and you don't want to change it then that's ok.

Reply:
OK.  I've added a class with some clearer test cases for these contingencies.

Reply:
Thank you